x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/UpdateSettingsStep.java

```diff
@@ -42,32 +57,16 @@ public void performAction(
         ClusterStateObserver observer,
         ActionListener<Void> listener
     ) {
-        UpdateSettingsRequest updateSettingsRequest = new UpdateSettingsRequest(indexMetadata.getIndex().getName()).masterNodeTimeout(
-            TimeValue.MAX_VALUE
-        ).settings(settings);
+        String indexName = targetIndexNameSupplier.apply(indexMetadata);
+        Settings settings = settingsSupplier.apply(indexMetadata);
+        UpdateSettingsRequest updateSettingsRequest = new UpdateSettingsRequest(indexName).masterNodeTimeout(TimeValue.MAX_VALUE)
+            .settings(settings);
         getClient(currentState.projectId()).admin()
             .indices()
             .updateSettings(updateSettingsRequest, listener.delegateFailureAndWrap((l, r) -> l.onResponse(null)));
     }
 
-    public Settings getSettings() {
-        return settings;
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(super.hashCode(), settings);
-    }
-
-    @Override
-    public boolean equals(Object obj) {
```

Comment:
There's not much value in overriding `hashCode` and `equals` when the only two fields are functions (which might be lambdas) and thus do not implement those methods.